lasagne/layers/special.py

```diff
@@ -15,6 +17,54 @@
 ]
 
 
+class LambdaLayer(Layer):
+    """
+    This layer provides boilerplate for a custom layer that applies a
+    simple transformation to the input.
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape.
+
+    function : callable
+        A function to be applied to the output of the previous layer.
+
+    output_shape : None, callable, or tuple
+        Specifies the output shape of this layer. If a tuple, this fixes the
+        output shape for any input shape (the tuple can contain None if some
+        dimensions may vary). If a callable, it should return the calculated
+        output shape given the input shape. If None, will attempt to determine
+        the output shape by applying the specified function.
+    """
+    def __init__(self, incoming, function, output_shape=None, **kwargs):
+        super(LambdaLayer, self).__init__(incoming, **kwargs)
+
+        if output_shape is None:
+            self._output_shape_memo = {}
+        else:
+            if hasattr(output_shape, '__call__'):
+                self.get_output_shape_for = output_shape
+            else:
+                self.get_output_shape_for = lambda _: tuple(output_shape)
```

Comment:
This is not pickable; we shouldn't use `lambda` in any of the classes we provide. (If a user chooses to pass a `lambda` expression for `function` or `output_shape`, it's their responsibility.)